src/repo/mod.rs

```diff
@@ -199,8 +199,8 @@ impl<TRepoTypes: RepoTypes> Repo<TRepoTypes> {
             .send(RepoEvent::UnprovideBlock(cid.to_owned()))
             .await
             .ok();
-        self.block_store.remove(cid).await?;
-        Ok(())
+
+        self.block_store.remove(cid).await
```

Comment:
I think we should add a TODO or FIXME to have `self.subscriptions.lock().await.cancel(cid)` here as well. I don't think we should do it right away since we don't have any way to communicate to bitswap we don't actually need the block after all.